torch/_dynamo/variables/tensor.py

```diff
@@ -744,74 +764,63 @@ def call_method(
-        # in `torch.overrides.get_default_nowrap_function()`. It's unclear how
-        # to do this easily in the current codebase since the resolution of
-        # `GetAttrVariable` depends on the type of the underlying object.
-
-        return TensorWithTFOverrideVariable(
-            unwrapped,
-            self.orig_tensor_variable_source,
-            self.subclass_torch_function__func,
-            self.subclass_type,
-        )
-
     def global_class_name(self):
         return f"__subclass_{self.subclass_type.__name__}"
 
     @staticmethod
-    def inline_torch_function_unwrapped(
-        tx,
-        original_func_var,
-        tensor_with_tf_override_source,
-        tf_func,
-        subclass_type,
-        options,
-        args,
-        kwargs,
-    ):
-        """
-        This function inlines the `__torch_function__` override for `original_func_var`.
-        For example, if the user code is
+    def can_dispatch_torch_function(tx, args, kwargs):
+        if tx.output.torch_function_enabled:
+            all_args = args + tree_flatten(kwargs)[0]
+            return any(
+                isinstance(arg, TensorWithTFOverrideVariable) for arg in all_args
+            )
+        else:
+            return False
 
-           x1 = torch.sigmoid(x0)
+    @staticmethod
+    def dispatch_torch_function(tx, fn, args, kwargs):
+        """Gathers all args that are TensorWithTFOverrideVariable and dispatches based on the ordering in _get_overloaded_args"""
 
-        And `x0` has an override, then:
-        * `original_func_var` will be a `VariableTracker` object wrapping `torch.sigmoid`
-        * `tensor_with_tf_override_source` will be the `Source` object from
-          the original tensor override instance in the beginning of the program
-        * `tf_func` will be the custom `__torch_function__` function
-        * `subclass_type` will be `type(x0)`
+        all_args = args + tree_flatten(kwargs)[0]
+        overloaded_args = _get_overloaded_args(
+            [arg for arg in all_args if isinstance(arg, TensorWithTFOverrideVariable)],
+            lambda x: x.subclass_type,
+        )
 
-        The caller is expected to properly massage args and kwargs before
-        passing them into this function.
+        for arg in overloaded_args:
+            res = arg.call_torch_function(
+                tx,
+                fn,
+                variables.TupleVariable(
+                    list({arg.subclass_type_var() for arg in overloaded_args})
+                ),
+                args,
+                kwargs,
+            )
 
-        The caller is responsible for wrapping the return value, if needed.
-        """
-        from . import UserDefinedClassVariable
-        from .builder import TupleVariable, VariableBuilder
+            if not (
+                isinstance(res, variables.ConstantVariable)
+                and res.value is NotImplemented
```

Comment:
Can you elaborate more with a behavioral example for this check?

Reply:
The canonical meaning of `x is y` is `IS_OP` bytecode definition

Reply:
I see, so basically use an IS_OP call to compare the result to a `NotImplemented` ConstantVariable? Fair enough, but it seems more of a style preference than providing tangible benefit, since we are throwing the error at compile time. Curious what voz thinks though too.